### ISSUE TITLE
`unnecessary_conversion_for_trait` false positive

### ISSUE THREAD

**Issue by smoelius** [trailofbits/dylint#1531]
`unnecessary_conversion_for_trait` recommends to remove the `iter` in the call to `extend`. But then `xs` would be unavailable for the call to `println!`.
```rs
fn main() {
    let xs = vec![[0u8; 16]];
    let mut ys: Vec<[u8; 16]> = Vec::new();
    ys.extend(xs.iter());
    //          ^^^^^^^
    println!("{:?}", xs);
}
```

**Comment by enfayz** [comment]
Hey @smoelius Could you assign this to me? I’d love to take it on.

**Comment by smoelius** [comment]
@enfayz Thank you very much for your interest.

As a first step, could I recommend reproducing the problem in a test?

The easiest approach would likely be to create new `.rs`, etc. files in this directory: https://github.com/trailofbits/dylint/tree/master/examples/supplementary/unnecessary_conversion_for_trait/ui

The test will need to be added to the lib.rs file, similar to here: https://github.com/trailofbits/dylint/blob/b83a03317dafa25c93455f57bb0c6ed206ab5140/examples/supplementary/unnecessary_conversion_for_trait/src/lib.rs#L425-L433

The test should have an `#[allow(unnecessary_conversion_for_trait)]` attribute somewhere inside of it, and should fail without that attribute.

If what I described is not too small, would you like to take it on?

**Comment by enfayz** [comment]
Yes, @smoelius 
I’d love to take this on! I'll create the test case and update lib.rs accordingly

**Comment by smoelius** [comment]
Great! I'm looking forward to your submission!

**Comment by enfayz** [comment]
@smoelius I've sent the PR #1531 Could you please have a look and provide feedback or suggestions if any changes are required
